src/client/datascience/constants.ts

```diff
@@ -18,18 +18,31 @@ export const KernelLauncherDaemonModule = 'vscode_datascience_helpers.kernel_lau
 export const PythonExtension = 'ms-python.python';
 
 // List of 'language' names that we know about. All should be lower case as that's how we compare.
+export const KnownKernelLanguageAliases = new Map<string, string>([
+    ['qsharp', 'q#'],
+    ['csharp', 'c#'],
```

Comment:
On my mac, `qsharp` is the langauge of the kernel.
Ie `q#` is not using the same pattern as `c$` & `f#`.
Added support for both, if there are such kernels.
With C++, we have xeus c++ using the version (added support for 11, 12 & 14).
Can't remember if there's a 12. Eitehr way doesn't matter.